src/librustc_typeck/constrained_type_params.rs

```diff
@@ -13,9 +13,14 @@ use rustc::ty::fold::{TypeFoldable, TypeVisitor};
 use rustc::util::nodemap::FnvHashSet;
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub enum Parameter {
-    Type(ty::ParamTy),
-    Region(ty::EarlyBoundRegion),
+pub struct Parameter(pub u32);
```

Comment:
Well, that's an internal conversion anyway...but I'd argue that's a problem with `ParamTy`. :)